controllers/certificatesigningrequest_controller.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2020 The cert-manager authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	capi "k8s.io/api/certificates/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/cert-manager/signer/api"
+	capihelper "github.com/cert-manager/signer/internal/kubernetes/apis/certificates/v1beta1"
+	"github.com/cert-manager/signer/internal/kubernetes/signer"
+)
+
+// CertificateSigningRequestSigningReconciler reconciles a CertificateSigningRequest object
+type CertificateSigningRequestSigningReconciler struct {
+	client.Client
+	Log           logr.Logger
+	Scheme        *runtime.Scheme
+	SignerName    string
+	Signer        *signer.Signer
+	EventRecorder record.EventRecorder
+}
+
+// +kubebuilder:rbac:groups=certificates.k8s.io.cert-manager.io,resources=certificatesigningrequests,verbs=get;list;watch
+// +kubebuilder:rbac:groups=certificates.k8s.io.cert-manager.io,resources=certificatesigningrequests/status,verbs=patch
+
+func (r *CertificateSigningRequestSigningReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("certificatesigningrequest", req.NamespacedName)
+	var csr capi.CertificateSigningRequest
+	if err := r.Client.Get(ctx, req.NamespacedName, &csr); client.IgnoreNotFound(err) != nil {
+		return ctrl.Result{}, fmt.Errorf("error %q getting CSR", err)
+	}
+	switch {
+	case !csr.DeletionTimestamp.IsZero():
+		log.Info("CSR has been deleted. Ignoring.")
+	case csr.Spec.SignerName == nil:
+		log.Info("CSR does not have a signer name. Ignoring.")
+	case *csr.Spec.SignerName != r.SignerName:
+		log.Info("CSR signer name does not match. Ignoring.", "signer-name", csr.Spec.SignerName)
```

Comment:
Let's make this `V(4)` to avoid spamming logs for every CSR that we want to ignore :) can we also put these cases _before_ the `case !csr.DeletionTimestamp.IsZero()` to avoid noise for signerNames we are ignoring?

Reply:
`V(4)` won't be logged, even in debug mode. See https://github.com/kubernetes-sigs/controller-runtime/issues/167

I'll use `V(1)` and add a `--debug-logging` flag, for now.